File: rust/day4_part1_2017.rs
use std::fs;

fn main() {
    let input = fs::read_to_string("input.txt").expect("Unable to read file");
    let valid_passphrases = input.lines().filter(|line| {
        let words: Vec<&str> = line.split_whitespace().collect();
        let mut unique_words = std::collections::HashSet::new();
        for word in words {
            if !unique_words.insert(word) {
                return false;
            }
        }
        true
    }).count();

    println!("{}", valid_passphrases);
}

File: swift/day4_part1_2017.swift

import Foundation

let fileURL = URL(fileURLWithPath: "input.txt")
let input = try String(contentsOf: fileURL)
let passphrases = input.components(separatedBy: .newlines)

let validPassphrases = passphrases.filter { passphrase in
    let words = passphrase.components(separatedBy: " ")
    return Set(words).count == words.count
}

print(validPassphrases.count)
